### PR TITLE
add DRA unit test: manager.go

### PR DESCRIPTION
**Pull request by charles-chenzz**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
add unit test for DRA manager
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by charles-chenzz** [comment]
```
W0603 08:02:58.615] Looking for address 'bootstrap-e2e-master-ip'
W0603 08:02:58.615] ERROR: (gcloud.compute.addresses.describe) Could not fetch resource:
W0603 08:02:58.826]  - The resource 'projects/k8s-infra-e2e-boskos-047/regions/us-central1/addresses/bootstrap-e2e-master-ip' was not found
W0603 08:02:58.826] 
W0603 08:02:58.826] Could not detect Kubernetes master node.  Make sure you've launched a cluster with 'kube-up.sh'
I0603 08:02:58.927] Project: k8s-infra-e2e-boskos-047
I0603 08:02:58.927] Network Project: k8s-infra-e2e-boskos-047
I0603 08:02:58.927] Zone: us-central1-b
I0603 08:02:58.927] Dumping logs from master locally to '/workspace/_artifacts'
I0603 08:02:58.927] Master not detected. Is the cluster up?
I0603 08:03:02.748] Dumping logs from nodes locally to '/workspace/_artifacts'
I0603 08:03:02.748] Detecting nodes in the cluster
I0603 08:03:02.748] No nodes found!
W0603 08:03:04.899] WARNING: The following filter keys were not present in any resource : name, zone
W0603 08:03:06.544] WARNING: The following filter keys were not present in any resource : name, zone
W0603 08:03:06.771] INSTANCE_GROUPS=
W0603 08:03:06.774] NODE_NAMES=
W0603 08:03:06.774] 2023/06/03 08:03:06 process.go:155: Step './cluster/log-dump/log-dump.sh /workspace/_artifacts' finished in 13.43640149s
W0603 08:03:06.774] 2023/06/03 08:03:06 node.go:53: Noop - Node Down()
W0603 08:03:06.775] 2023/06/03 08:03:06 process.go:96: Saved XML output to /workspace/_artifacts/junit_runner.xml.
W0603 08:03:07.066] 2023/06/03 08:03:06 process.go:153: Running: bash -c . hack/lib/version.sh && KUBE_ROOT=. kube::version::get_version_vars && echo "${KUBE_GIT_VERSION-}"
W0603 08:03:07.066] 2023/06/03 08:03:07 process.go:155: Step 'bash -c . hack/lib/version.sh && KUBE_ROOT=. kube::version::get_version_vars && echo "${KUBE_GIT_VERSION-}"' finished in 291.539794ms
W0603 08:03:07.081] 2023/06/03 08:03:07 main.go:328: Something went wrong: encountered 1 errors: [error during go run /go/src/k8s.io/kubernetes/test/e2e_node/runner/remote/run_remote.go --cleanup -vmodule=*=4 --ssh-env=gce --results-dir=/workspace/_artifacts --project=k8s-infra-e2e-boskos-047 --zone=us-central1-b --ssh-user=prow --ssh-key=/workspace/.ssh/google_compute_engine --ginkgo-flags=--focus="\[NodeFeature:DynamicResourceAllocation\]" --skip="\[Flaky\]" --test_args=--feature-gates="DynamicResourceAllocation=true" --service-feature-gates="DynamicResourceAllocation=true --runtime-config=resource.k8s.io/v1alpha2=true --container-runtime-endpoint=unix:///run/containerd/containerd.sock --container-runtime-process-name=/usr/bin/containerd --container-runtime-pid-file= --kubelet-flags="--cgroup-driver=systemd --cgroups-per-qos=true --cgroup-root=/ --runtime-cgroups=/system.slice/containerd.service" --extra-log="{\"name\": \"containerd.log\", \"journalctl\": [\"-u\", \"containerd\"]}" --test-timeout=1h5m0s --image-config-file=/workspace/test-infra/jobs/e2e_node/containerd/containerd-release-1.7-presubmit/image-config-presubmit.yaml: exit status 1]
W0603 08:03:07.084] Traceback (most recent call last):
W0603 08:03:07.084]   File "/workspace/./test-infra/jenkins/../scenarios/kubernetes_e2e.py", line 723, in <module>
W0603 08:03:07.084]     main(parse_args())
W0603 08:03:07.084]   File "/workspace/./test-infra/jenkins/../scenarios/kubernetes_e2e.py", line 569, in main
W0603 08:03:07.085]     mode.start(runner_args)
W0603 08:03:07.085]   File "/workspace/./test-infra/jenkins/../scenarios/kubernetes_e2e.py", line [228](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/118436/pull-kubernetes-node-e2e-containerd-dra/1664904054897119232/#1:build-log.txt%3A228), in start
W0603 08:03:07.085]     check_env(env, self.command, *args)
W0603 08:03:07.085]   File "/workspace/./test-infra/jenkins/../scenarios/kubernetes_e2e.py", line 111, in check_env
W0603 08:03:07.085]     subprocess.check_call(cmd, env=env)
W0603 08:03:07.085]   File "/usr/lib/python3.9/subprocess.py", line 373, in check_call
W0603 08:03:07.085]     raise CalledProcessError(retcode, cmd)
W0603 08:03:07.096] subprocess.CalledProcessError: Command '('kubetest', '--dump=/workspace/_artifacts', '--gcp-service-account=/etc/service-account/service-account.json', '--up', '--down', '--test', '--deployment=node', '--provider=gce', '--cluster=bootstrap-e2e', '--gcp-network=bootstrap-e2e', '--gcp-zone=us-central1-b', '--node-test-args=--feature-gates="DynamicResourceAllocation=true" --service-feature-gates="DynamicResourceAllocation=true --runtime-config=resource.k8s.io/v1alpha2=true --container-runtime-endpoint=unix:///run/containerd/containerd.sock --container-runtime-process-name=/usr/bin/containerd --container-runtime-pid-file= --kubelet-flags="--cgroup-driver=systemd --cgroups-per-qos=true --cgroup-root=/ --runtime-cgroups=/system.slice/containerd.service" --extra-log="{\\"name\\": \\"containerd.log\\", \\"journalctl\\": [\\"-u\\", \\"containerd\\"]}"', '--node-tests=true', '--test_args=--focus="\\[NodeFeature:DynamicResourceAllocation\\]" --skip="\\[Flaky\\]"', '--timeout=65m', '--node-args=--image-config-file=/workspace/test-infra/jobs/e2e_node/containerd/containerd-release-1.7-presubmit/image-config-presubmit.yaml')' returned non-zero exit status 1.
E0603 08:03:07.096] Command failed
I0603 08:03:07.096] process 534 exited with code 1 after 2.4m
E0603 08:03:07.097] FAIL: pull-kubernetes-node-e2e-containerd-dra
I0603 08:03:07.097] Call:  gcloud auth activate-service-account --key-file=/etc/service-account/service-account.json
W0603 08:03:07.902] Activated service account credentials for: [prow-build@k8s-infra-prow-build.iam.gserviceaccount.com]
I0603 08:03:08.044] process 9664 exited with code 0 after 0.0m
I0603 08:03:08.044] Call:  gcloud config get-value account
I0603 08:03:08.879] process 9674 exited with code 0 after 0.0m
I0603 08:03:08.880] Will upload results to gs://kubernetes-jenkins/pr-logs using prow-build@k8s-infra-prow-build.iam.gserviceaccount.com
I0603 08:03:08.880] Upload result and artifacts...
I0603 08:03:08.880] Gubernator results at https://gubernator.k8s.io/build/kubernetes-jenkins/pr-logs/pull/118436/pull-kubernetes-node-e2e-containerd-dra/1664904054897119232
I0603 08:03:08.880] Call:  gsutil ls gs://kubernetes-jenkins/pr-logs/pull/118436/pull-kubernetes-node-e2e-containerd-dra/1664904054897119232/artifacts
W0603 08:03:10.[231](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/118436/pull-kubernetes-node-e2e-containerd-dra/1664904054897119232/#1:build-log.txt%3A231)] CommandException: One or more URLs matched no objects.
E0603 08:03:10.471] Command failed
I0603 08:03:10.471] process 9684 exited with code 1 after 0.0m
```


the failed test seems not cause by the code that I add, have no idea why

File: pkg/kubelet/cm/dra/manager_test.go [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+	_, err = fakeKubeClient.ResourceV1alpha2().ResourceClaims(pod.Namespace).Create(context.Background(), resourceClaim, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("failed to create ResourceClaim: %+v", err)
+	}
+
+	// Call the PrepareResources method
+	err = manager.PrepareResources(pod)
+	if err != nil {
+		t.Fatalf("PrepareResources failed: %+v", err)
+	}
```
Comment (bart0sh) [inline pkg/kubelet/cm/dra/manager_test.go:147]: This looks strange. How `PrepareResources` can possibly succeed if DRA Kubelet plugin is not running? There must be a bug somewhere in the code.

**Comment by bart0sh** [comment]
@charles-chenzz
> the failed test seems not cause by the code that I add

Correct. Please, ignore.

**Comment by bart0sh** [comment]
@charles-chenzz please fix `pull-kubernetes-verify` failure, thanks.

**Comment by charles-chenzz** [comment]
@bart0sh CI test passed, PTAL when you have time

**Comment by bart0sh** [comment]
@charles-chenzz Can you also address [this comment](https://github.com/kubernetes/kubernetes/pull/118436#pullrequestreview-1461133191) ?

File: pkg/kubelet/cm/dra/manager_test.go (hunk 1 above)
Comment (charles-chenzz) [reply pkg/kubelet/cm/dra/manager_test.go:147]: #118468 this issues is created to track this place? or it's different problem

Comment (bart0sh) [reply pkg/kubelet/cm/dra/manager_test.go:147]: I think it's a different problem. TestPrepareResources doesn't test anything. I think it silently returns at this place:
```	cache, err := newClaimInfoCache(t.TempDir(), "")
	if err != nil {
		return
	}
```
If you replace `return` with `t.Fatalf` call, you'll see it.

Comment (bart0sh) [reply pkg/kubelet/cm/dra/manager_test.go:147]: You can also look at the test coverage to see how manager.go code is covered by the tests. In my setup coverage viewer shows 13.3% coverage for manager.go, which means most of the code is not tested.

Comment (charles-chenzz) [reply pkg/kubelet/cm/dra/manager_test.go:147]: will check this later today. can you review the rest of the code and point out where do I need to change/address as well? would solve it when I back to home. Thanks!

Comment (bart0sh) [reply pkg/kubelet/cm/dra/manager_test.go:147]: Overall observation after briefly looking at the code - you should avoid bare `return` on errors. Use either `t.Fatalf` or `t.Errorf` depending on severity of the failure.  `return` just hides the error.

**Comment by TommyStarK** [comment]
/close 
see #118711

**Pull request closed without merging**